### PR TITLE
RHBPMS-4586: Unable to create container with name which includes underscore (_)

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/RHBPMS-4586

### CODE REVIEW

**Comment by karreiro** [comment]
@manstis Could you take a look? :-)

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/SpecManagementServiceCDI.java [hunk 1]
```diff
@@ -108,13 +109,19 @@ private String validContainerIdWithSuffix( final String serverTemplateId,
...
+            if ( !( Character.isLetterOrDigit( c ) || allowedChar( c ) ) ) {
                 return false;
             }
         }
+
         return true;
     }
+
+    private boolean allowedChar( final char c ) {
+        final char[] allowedChars = { ':', '-', '.', '_' };
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/SpecManagementServiceCDI.java]: Ouch; do we have to construct this array each time the method is called (from a loop, that iterates the same number of times as identifier is characters long)...?

Comment (karreiro) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-backend/src/main/java/org/kie/workbench/common/screens/server/management/backend/service/SpecManagementServiceCDI.java]: I was thinking just about readability, but you're right, Mike, it does not worth in this case.

**Comment by karreiro** [comment]
Thanks, Mike! :-)

**Comment by jomarko** [comment]
@karreiro thanks for test covering a lot of 'special' characters.

**Pull request merged**